### PR TITLE
Front end - http.

### LINKED ISSUE #11: front end
http

- /get
- /put

### PR DESCRIPTION
**Pull request by ysono**
```sh
curl -i -X PUT localhost:3000/put/mykey -d myvalue
curl -i localhost:3000/get/mykey
```

Do you have a preferred web framework? `routerify` was the first one that worked for me.

Now that concurrency is introduced, next is figuring out how a background job can fit in.

### CODE REVIEW

File: src/frontend/http.rs [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+use routerify::{RequestInfo, Router, RouterService};
+use std::sync::{Arc, RwLock};
+use std::{convert::Infallible, net::SocketAddr};
+
+async fn get_handler(req: Request<Body>) -> Result<Response<Body>, Infallible> {
+    let key: &String = req.param("key").unwrap();
+    let key = Key::from(key.clone());
+
+    let lsm = req.data::<Arc<RwLock<LSM>>>().unwrap();
+    let lsm = lsm.read().unwrap();
```
Comment (btc) [inline src/frontend/http.rs:26]: where is `lsm` instantiated?

**Review by btc — COMMENTED** [review]
curious where LSM is instantiated?

File: src/frontend/http.rs (hunk 1 above)
Comment (ysono) [reply src/frontend/http.rs:26]: https://github.com/ysono/pancake/pull/15/files#diff-51f622a727225cad9caed61cc103982c3a45ec6227624abe42a441d9c79d1ab3R61 here, during `Router::build()`.

Also, I don't think this needs to be a RwLock. Just Arc.

Comment (btc) [reply src/frontend/http.rs:26]: ah i see!

Comment (ysono) [reply src/frontend/http.rs:26]: Actually we need RwLock.

**Comment by btc** [comment]
> Do you have a preferred web framework? routerify was the first one that worked for me.

`warp` is the one i intend to try in potential Rust projects that require an HTTP API. but i dont know anything about them. no basis for comparison.

https://github.com/seanmonstar/warp

**Review by btc — APPROVED** [review]

**Comment by ysono** [comment]
Oh oops forgot delete lol. Another PR.

**Pull request merged**